### ISSUE TITLE
*-fields.csv files no longer exist, but need to be downloaded

### ISSUE THREAD

**Issue by seemantk** [cfinst/cfinst.github.io#123]
Following up on the Jekyllization, since the field info is now in [_data/sections](https://github.com/cfinst/cfinst.github.io/tree/master/_data/sections) in the appropriate `controls.yml` file, the *-fields.csv files in [/data](https://github.com/cfinst/cfinst.github.io/commit/db9e28ca94a17a9a09e532496d6d86742a969d59) are now obsolete, as data entry should be done via the github interface.

Let's ideate on how to make the info available to data downloaders.

**Comment by seemantk** [comment]
One idea could be to use Jekyll templates to generate each one.  However, they then couldn't be part of the same zip download, since that is done at commit time, while these files would be generated post-commit.  So, perhaps a supplemental zip download that gets assembled on the fly?

Thus:

1. User clicks download
1. Packaged zip file containing database dump is transmitted
1. Generate a new zip file containing the *-fields.csv files in them

Alternatively:

1. User loads page
2. Generate supplemental zip file and cache it (or stash it in the DOM somewhere)
3. User clicks download
  * Send package zip
  * Send cached zip

**Comment by curran** [comment]
@seemantk I like the idea of using Jekyll to generate the *-fields.csv files,then bundling them into the single download .zip. Do you think that's a viable route? It would require running Jekyll before the packaging step.

**Comment by curran** [comment]
Maybe the path of least resistance would be to go back to using those .csv files as the source data, like we were before? Not sure..

**Comment by seemantk** [comment]
This is the list of fields in the *-fields.csv files:

- Field Name
- Value Type
- Question on Data Entry Form
 - Short Label
 - Additional Notes

Since my original comment, the Jekyllization actually stores all those values on the DOM itself ([within the data- attributes of the options in the dropdowns](https://github.com/cfinst/cfinst.github.io/blob/master/_includes/questions/question-answer.html#L7)).

Thus, we _should_ be able to generate the appropriate -fields.csv files on the fly.

**Comment by curran** [comment]
Aha! That's excellent. We can use the same logic from [downloadCSV](https://github.com/cfinst/cfinst.github.io/blob/db0dd75caeaf53c353f6e508c7d8881e7bb5e041/js/schematic.js#L229).

With this approach, there are twi significant downsides:

 * These CSVs will not end up in the .zip file bundle.
 * It will not be possible to link into this CSV download from a page within the CFI CMS.

**Comment by curran** [comment]
This is the only issue blocking project completion and final delivery at this point.

Some thoughts on how this relates to requirements and potential solutions:

 * The end product will consist of two pages in the CFI CMS - the "landing page" and the "data download page", and one externally hosted page via GitHub Pages, the "visualization page".
 * The "data download page" needs to be able to provide links to the data downloads, hosted via GitHub pages, *without* opening the "visualization page".
 * Therefore, the CSV files containing the field descriptions *must* be computed as part of the data packaging step (described in the README under https://github.com/cfinst/cfinst.github.io#packaging-download-files) and included in the resulting .zip files.

Given that, I see the following options as possible solutions:

 * Running `jekyll build` as part of the packaging step, where we could have the field descriptions CSV generated using a Jekyll template (I think it's safe to assume that whoever is running the packaging step will have a working Jekyll installation).
 * Reverting back to using the CSV files as the "source of truth" for field descriptions, and removing these from the Jekyll config.

@seemantk Which of these options would you prefer?

**Comment by seemantk** [comment]
@curran, please see #175 for CSV files made available in download/metadata/

I'm currently looking at https://stuk.github.io/jszip/ to see if we can dynamically zip them up for download.

**Comment by curran** [comment]
@seemantk Would the jszip approach enable linking to the download from the "data download page" residing in the CFI CMS?

**Comment by seemantk** [comment]
@curran yes, I should think so.

But before we go down that road, looking at [the packaging instructions](https://github.com/cfinst/cfinst.github.io#packaging-download-files), looks like the packaging script can skip copying the *-fields.csv over, because those files are already in downloads/metadata.

tl;dr the packaging script in bin/ should be able to work now with a little bit of modification

**Comment by seemantk** [comment]
I reckon the following changes in the packaging script should do it:

1. run `jekyll build`
2. use `_site/downloads/metadata as the source directory for the *-fields.csv files

**Comment by curran** [comment]
Agreed, the copy step is no longer required.

**Comment by curran** [comment]
It occurs to me that we could easily just have a single .zip file to download, with subdirectories for `data` (raw tables) and `metadata` (field descriptions)`. I think I'll go down that road and iterate the packaging script.
